### DISCUSSION TITLE
컴퓨터 네트워크 (재민 -> 가은)

### DISCUSSION THREAD

**Discussion by dku19jam** [tempdku/temp#10] [General]
1. 가은이는 학생회 행사 공지 프로그램을 만드려고 한다. 주요 기능으로는 채팅, 이메일 전송, 파일 업로드, 동영상 재생 등이 있다. 각각의 기능들을 전송계층의 어느 프로토콜을 사용할 것이고 그 이유는 무엇인지 설명해 보세요.
2. 가은씨는 프론트엔드 직무를 희망하신다고 들었는데 혹시 RESTful 한 API에 대한 내용을 어느정도 알고 계신가요? RESTful API를 개발하는 방법들은 무엇이 있을까요?

**Comment by gaeunnlee** [comment]
1. UDP는 비연결형 프로토콜로, 연결을 설정하고 해제하는 과정이 존재하지 않으므로 대역폭과 대기 시간이 적어, 특정 시간에 대규모 사용자가 정보를 제공하거나 트래픽을 관리하고 응답을 해야할 때, 게임이나 비디오 스트리밍, 온라인 회의 등에 분야에서 UDP 기술을 사용합니다. TCP는 데이터를 안정적으로 전달하기 위해 신뢰성 있는 통신을 제공하므로, 채팅, 이메일 전송, 파일 업로드 등 신뢰성이 중요한 응용 프로그램에 사용됩니다.

2. REST는 HTTP 통신을 활용하기 위해 고안된 아키텍처로, HTTP 메서드를 사용해 자원에 대한 연산을 처리하는 방식입니다. REST는 클라이언트와 서버 간 요청과 응답을 주고 받는 독립적인 구조를 가지며, 자원, 행위, 표현으로 구성되어 있는 REST API를 통하여 요청을 표현합니다. RESTful한 API를 개발하기 위해서 HTTP 메서드는 명시적이어야 합니다. 그리고 URL 경로는 슬래시로 계층 관계를 표현하며, URI 마지막에 슬래시가 들어가면 안 됩니다. URI 경로에는 언더바를 사용하면 안 되고, 소문자 사용을 지향해야 합니다.
